src/main/java/baron/tasks/TaskManager.java

```diff
@@ -0,0 +1,271 @@
+                }
+            } else {
+                throw new BaronException("Please specify only one deadline.");
+            }
+        } else {
+            String[] splitString = commandArg.split(" /at ", 2);
+            if (splitString.length == 2) {
+                LocalDateTime localDateTime = DateTimeUtil.getDateTime(splitString[1]);
+                newTask = new Event(splitString[0], localDateTime);
+            } else if (splitString.length == 1) {
+                if (splitString[0].startsWith("/at")) {
+                    throw new BaronException(Message.generateEmptyDescMessage(taskType));
+                }
+                throw new BaronException("Please specify a date by the /at keyword.");
+            } else {
+                throw new BaronException("Please specify only one date.");
+            }
+        }
+
+        this.taskList.add(newTask);
+        this.previousTaskList = this.getAllTasks();
+        return newTask;
+    }
+
+    /**
+     * Retrieves the {@code Task} at the specified position (1-indexed).
+     *
+     * @param index the index of the task (1-indexed).
+     * @return the {@code Task} at the specified position (1-indexed).
+     * @throws BaronException If there is no tasks or index is out of bound.
+     */
+    public Task getTask(int index) throws BaronException {
+        if (this.getTaskCount() == 0) {
+            throw new BaronException(Message.MESSAGE_NO_TASK);
+        } else if (index < 1 && index > this.getTaskCount()) {
+            throw new BaronException(Message.generateTaskIndexOutOfBoundMessage(this.getTaskCount()));
+        }
+        return this.taskList.get(index - 1);
+    }
+
+    /**
+     * Retrieves all tasks in this {@code TaskManager} in the form of {@code ArrayList<Task>}.
+     *
+     * @return a new {@code ArrayList<Task>} containing all the tasks in this {@code TaskManager}.
+     */
+    public ArrayList<Task> getAllTasks() {
+        return new ArrayList<>(this.taskList);
+    }
+
+    /**
+     * Marks the task that is at the specified position (1-indexed).
+     *
+     * @param index the index of the task (1-indexed).
+     * @return true if the task is marked successfully.
+     * @throws BaronException If there is no task, the specified task was already marked,
+     *                        or the index is out of bound.
+     */
+    public boolean markTask(int index) throws BaronException {
+        if (this.isEmpty()) {
+            throw new BaronException(Message.MESSAGE_NO_TASK);
+        } else if (index > 0 && index <= this.getTaskCount()) {
+            if (this.taskList.get(index - 1).mark()) {
+                return true;
+            } else {
+                throw new BaronException("The task is already marked.");
+            }
+        } else {
+            throw new BaronException(Message.generateTaskIndexOutOfBoundMessage(this.getTaskCount()));
+        }
+    }
```

Comment:
Maybe you can use guard clauses to make happy path more prominent? This can be applied for other methods in this class as well.

Reply:
Hi, can you provide me with an example of a guard clause? :)